aten/src/ATen/native/cuda/ForeachFunctors.cuh

```diff
@@ -10,6 +10,14 @@ template<typename scalar_t> struct get_opmath_t { using opmath_t = scalar_t; };
 template<> struct get_opmath_t<at::Half> { using opmath_t = float; };
 template<> struct get_opmath_t<at::BFloat16> { using opmath_t = float; };
 
+template<typename scalar_t> struct get_promoted_t { using type = scalar_t; };
+template<> struct get_promoted_t<bool> { using type = float; };
+template<> struct get_promoted_t<int8_t> { using type = float; };
+template<> struct get_promoted_t<uint8_t> { using type = float; };
+template<> struct get_promoted_t<int16_t> { using type = float; };
+template<> struct get_promoted_t<int32_t> { using type = float; };
+template<> struct get_promoted_t<int64_t> { using type = float; };
+
```

Comment:
Could `at::promote_types` be used here or do we want to use this mapping here?

Reply:
But, for binary & pointwise ops, I guess https://github.com/crcrpar/pytorch/blob/master/c10/core/ScalarType.h#L365 would be helpful.